test/integration.test.js

```diff
@@ -1,142 +1,193 @@
 import test from 'ava';
 import pify from 'pify';
-import SemanticReleaseError from '@semantic-release/error';
 import releaseNotesGenerator from '../lib/index';
```

Comment:
how does this work? Is this transpiled by Ava before running? I think I'd avoid that syntax and do `const name = require('<path>')`?

Reply:
Yes, Ava transpile using Babel so we can write test in ES6 independently of what node version we are running on and supporting. It transpile only the test, not the sources.
It's transpiled to `const name = require('<path>')` when running test.
I'd rather keep the es6 syntax whenever we can as it's more standard. Plus I'm not sure it we want to modify that in every test file across all the repo...

Reply:
with es6 syntax you mean `const name = require('<path>')`? The import syntax is not valid es6 is it?

Reply:
No, by es6 syntax, I mean `import name from 'path'`.
Node require to use `require` (no pun intended) because it doesn't support es6 module yet. I think it's planned for Node 10.

Reply:
what benefit does the import syntax have? I'd avoid transpiling altogether. release-notes-generator requires node 8, we can just make it run in that, the less build tools, the better. I'd vote to disable able altogether https://github.com/avajs/ava#configuration I can only imagine if errors are thrown then the transpilation step makes debugging more complicated

Reply:
Import syntax doesn't have any technical benefit, it's just more standard.

Ava compile only the test files, not the source files. That allow to use any new javascript feature. 
There is not really and additional build tools per say, as it's all handled by Ava transparently.

All errors thrown reference the exact lines of your tests file and source code. All the stack traces reference only the line of code in the tests file and source code, no additional stuff due to transpilation.

It's all completely transparent, doesn't touch the source code (the source tested are the exact sources published) and allow to use latest JS features to write test (so we can write test in the future that use JS features not supported in Node 8 for example).

Reply:
I'm worried about too much "magic" :) I just bit me too often.

I'm also worried that code in the tests looks different from the code in source files. If a new contributor would come in they might try to use the "import" syntax in the source files because they are used in tests, or might just get confused why they are used in one but not the other.

> write test in the future that use JS features not supported in Node 8

this is not a situation we have right now. Maybe some super cool feature becomes available and then we can reassess, but for now I would suggest we don't use the import syntax. I don't mind to leave the transpilation, but I think it'd be helpful to keep test and source code syntax the same

Reply:
> I'm worried about too much "magic" :) I just bit me too often.

Yes. But both Babel and Ava are mature and well maintained and used by a lot of projects. Pretty much every project that publish a node module use Babel to support Node 4. So I think it's pretty safe.
And to some extend that "magic", nice cutting edge tools, new syntax and features, etc... is what make developing open-source in JS fun. At least to me :)

> If a new contributor would come in they might try to use the "import" syntax in the source files because they are used in tests, or might just get confused why they are used in one but not the other.

We could mention that in a  CONTRIBUTING file on the main repo.
It's pretty obvious when you open the sources that the tests use `import` and the sources use `require`. I'm not sure why someone would suddenly start using `import` in a file that already contains some `require`. And if they do, Node would throw a syntax error refusing the keyword `import`.

I rather spend the time to document and explain things to contributors via comments in PRs and issues, rather than trying to "hide/avoid" some JS weirdness or shortcoming (like having 2 ways to import module because Node is not ES6 module compliant yet).

As a side note, using `babel-register` for `semantic-release` would allow to use the `import` syntax everywhere.

> but for now I would suggest we don't use the import syntax.

There is dozen and dozen of test files using that syntax now across all semantic-release repo, not sure it's the best usage of our time to change that....

Reply:
okay don't worry about it, let's see if people run into this at all :)